ci/Jenkinsfiles/export.groovy

```diff
@@ -0,0 +1,329 @@
+                // Only used with jx preview on pr branches
+                String noCommentOpt = '';
+                if (nsExists) {
+                  noCommentOpt = '--no-comment'
+                  // Previous preview deployment needs to be scaled to 0 to be replaced correctly
+                  sh "kubectl --namespace ${PREVIEW_NAMESPACE} scale deployment export --replicas=0"
+                } else {
+                  sh "kubectl create namespace ${PREVIEW_NAMESPACE}"
+                }
+                sh "kubectl --namespace platform get secret kubernetes-docker-cfg -ojsonpath='{.data.\\.dockerconfigjson}' | base64 --decode > /tmp/config.json"
+                sh """kubectl create secret generic kubernetes-docker-cfg \
+                    --namespace=${PREVIEW_NAMESPACE} \
+                    --from-file=.dockerconfigjson=/tmp/config.json \
+                    --type=kubernetes.io/dockerconfigjson --dry-run -o yaml | kubectl apply -f -"""
+                String previewCommand = "jx step helm install --namespace ${PREVIEW_NAMESPACE} --name ${PREVIEW_NAMESPACE} --verbose ."
+
+                // third build and deploy the chart
+                // waiting for https://github.com/jenkins-x/jx/issues/5797 to be fixed in order to remove --source-url
+                sh """
+                  jx step helm build --verbose
+                  mkdir target && helm template . --output-dir target
+                  ${previewCommand}
+                """
+
+                String nuxeoUrl = "export.${PREVIEW_NAMESPACE}.svc.cluster.local/nuxeo"
+                String explorerUrl = "${nuxeoUrl}/site/distribution"
+                echo """
+                ----------------------------------------
+                Perform export on ${explorerUrl}
+                ----------------------------------------
+                """
+                String distribId = "${params.SNAPSHOT_NAME}-${NUXEO_IMAGE_VERSION}".replaceAll(" ", "%20")
+                String curlCommand = "curl --user Administrator:Administrator ${CURL_OPTIONS}"
+                sh """
+                  kubectl rollout status deployment export \
+                    --timeout=3m \
+                    --namespace=${PREVIEW_NAMESPACE}
+
+                  ${curlCommand} ${explorerUrl} --output home.html
+                  ${curlCommand} -d 'name=${params.SNAPSHOT_NAME}' ${explorerUrl}/save
+                  ${curlCommand} ${explorerUrl}/download/${distribId} --output export.zip
+                """
+                if (params.PERFORM_JSON_EXPORT) {
+                  sh """
+                    ${curlCommand} ${explorerUrl}/${distribId}/json --output export.json
+                  """
+                }
+              } finally {
+                // cleanup jx namespace
+                sh """
+                  jx step helm delete export \
+                    --namespace=${PREVIEW_NAMESPACE} \
+                    --purge \
+                    --ignore-not-found=true
```

Comment:
should be added to the `kubectl delete ns` command instead

Reply:
oops :+1: